### PR TITLE
RSDK-8883: Have package downloading better interpret file does not exist errors.

### PR DESCRIPTION
**Pull request by dgottlieb**
I split this into two commits:
* The [logging logic](https://github.com/viamrobotics/rdk/pull/4402/commits/46ce76f4431a636dfa60de2fd28fdb9dc32c8deb) `packageIsSynced`
* [Migrating from](https://github.com/viamrobotics/rdk/pull/4402/commits/ee69673c954ab4f4e74f3dd5b2b4c0b7e91614b8) `pkg/errors.Wrapf` to stdlib `fmt.Errorf("...%w", err)`

Old log:
```
2024-09-24T13:16:10.175-0400	DEBUG	packages/utils.go:312	unchecked error	{"error": "cannot find /home/dgottlieb/.viam/packages/data/module/4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4-nicksensor-0_0_1-linux-amd64.status.json: open /home/dgottlieb/.viam/packages/data/module/4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4-nicksensor-0_0_1-linux-amd64.status.json: no such file or directory", "errorVerbose": "open /home/dgottlieb/.viam/packages/data/module/4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4-nicksensor-0_0_1-linux-amd64.status.json: no such file or directory\ncannot find /home/dgottlieb/.viam/packages/data/module/4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4-nicksensor-0_0_1-linux-amd64.status.json\ngo.viam.com/rdk/robot/packages.readStatusFile\n\t/home/dgottlieb/viam/rdk/robot/packages/utils.go:342\ngo.viam.com/rdk/robot/packages.packageIsSynced\n\t/home/dgottlieb/viam/rdk/robot/packages/utils.go:310\ngo.viam.com/rdk/robot/packages.packagesAreSynced\n\t/home/dgottlieb/viam/rdk/robot/packages/utils.go:325\ngo.viam.com/rdk/robot/packages.(*deferredPackageManager).getManagerForSync\n\t/home/dgottlieb/viam/rdk/robot/packages/deferred_package_manager.go:123\ngo.viam.com/rdk/robot/packages.(*deferredPackageManager).Sync\n\t/home/dgottlieb/viam/rdk/robot/packages/deferred_package_manager.go:81\ngo.viam.com/rdk/robot/impl.(*localRobot).reconfigure\n\t/home/dgottlieb/viam/rdk/robot/impl/local_robot.go:1200\ngo.viam.com/rdk/robot/impl.(*localRobot).Reconfigure\n\t/home/dgottlieb/viam/rdk/robot/impl/local_robot.go:1168\ngo.viam.com/rdk/robot/impl.newWithResources\n\t/home/dgottlieb/viam/rdk/robot/impl/local_robot.go:571\ngo.viam.com/rdk/robot/impl.New\n\t/home/dgottlieb/viam/rdk/robot/impl/local_robot.go:595\ngo.viam.com/rdk/web/server.(*robotServer).serveWeb\n\t/home/dgottlieb/viam/rdk/web/server/entrypoint.go:373\ngo.viam.com/rdk/web/server.(*robotServer).runServer\n\t/home/dgottlieb/viam/rdk/web/server/entrypoint.go:193\ngo.viam.com/rdk/web/server.RunServer\n\t/home/dgottlieb/viam/rdk/web/server/entrypoint.go:173\ngo.viam.com/utils.contextualMain[...]\n\t/home/dgottlieb/viam/rdk/vendor/go.viam.com/utils/runtime.go:72\ngo.viam.com/utils.ContextualMain[...]\n\t/home/dgottlieb/viam/rdk/vendor/go.viam.com/utils/runtime.go:29\nmain.main\n\t/home/dgottlieb/viam/rdk/web/cmd/server/main.go:19\nruntime.main\n\t/home/dgottlieb/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.13.linux-amd64/src/runtime/proc.go:267\nruntime.goexit\n\t/home/dgottlieb/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.13.linux-amd64/src/runtime/asm_amd64.s:1650"}
```

New Log (for fs.NotExists):
```
2024-09-30T14:24:05.621Z	INFO	rdk.package_manager	packages/utils.go:313	New package to download detected	{"name":"m_ncs_nicksensor","version":"0.0.1-linux-amd64","id":"4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4/nicksensor"}
```

New Log (if an unknown error):
```
2024-09-30T14:05:34.905Z	WARN	rdk.package_manager	packages/utils.go:316	Error reading status file for package	{"packageName":"m_ncs_nicksensor","packageVersion":"0.0.1-linux-amd64","packageId":"4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4/nicksensor","packagesDir":"/home/dgottlieb/.viam/packages","err":"cannot find \"/home/dgottlieb/.viam/packages/data/module/4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4-nicksensor-0_0_1-linux-amd64.status.json\" open /home/dgottlieb/.viam/packages/data/module/4593c406-dda4-40ff-9ea2-e5eb2fbe4ac4-nicksensor-0_0_1-linux-amd64.status.json: no such file or directory"}
```

### CODE REVIEW

**Review by abe-winter — APPROVED** [review]
did not know about `%q` or `%w`

File: robot/packages/utils.go [hunk 1]
```diff
@@ -182,18 +182,18 @@ func unpackFile(ctx context.Context, fromFile, toDir string) error {
 			// Ex: tar -czf package.tar.gz ./bin/module.exe
 			parent := filepath.Dir(path)
 			if err := os.MkdirAll(parent, 0o700); err != nil {
-				return errors.Wrapf(err, "failed to create directory %q", parent)
```
Comment (dgottlieb) [inline robot/packages/utils.go:185]: I saw this one using %q while adjacent ones were using %s, so I figured standardizing on quoted strings was preferable. Happy to undo.

**Review by michaellee1019 — APPROVED** [review]
LGTM! TY for improving. It might be "too much info" to record each package status sync in the logs as INFO, when it was previously DEBUG only. But I think lets only move back to DEBUG if there are complaints that logs are too much.

**Comment by dgottlieb** [comment]
Those errors represent that we have a new package to download. We should instead 
log that intention.

**Pull request merged**